Src/GoogleApis.Auth.DotNet4/OAuth2/ServiceAccountCredential.cs

```diff
@@ -72,6 +76,20 @@ public Initializer(string id, string tokenServerUrl) : base(tokenServerUrl)
                 Scopes = new List<string>();
             }
 
+            /// <summary>Constructs a new initializer using the given id and the private key in the ClientCredentialParameters.</summary>
+            public Initializer(ClientCredentialParameters clientCredentialParameters)
+                : this(clientCredentialParameters.ClientEmail, GoogleAuthConsts.TokenUrl)
+            {
+                Utilities.ThrowIfNullOrEmpty(clientCredentialParameters.ClientEmail, "ClientEmail");
+                Utilities.ThrowIfNullOrEmpty(clientCredentialParameters.Base64PrivateKey, "PrivateKey");
+
+                var privateKeyBytes = Convert.FromBase64String(clientCredentialParameters.Base64PrivateKey);

```

Comment:
I'd be leaning towards extracting a helper method call sth like "ConvertPkcs8ToRsaParameters", which would take the private key in the well known text format (including the ---BEGIN PRIVATE KEY---  and ---END PRIVATE KEY---- header and trailer) and returns an RSAParameters instance.

In the summary of the method you can then explain that this converts from the PKCS8 format into RSAParameters that are digestible by the rest of the library. As there is no support for reading PKCS8 format in .NET, you are using BouncyCastles.

That way, the dependency on BouncyCastles is properly documented and no one will wonder why you need it for.